atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -37,41 +37,123 @@
 
 /**
  * Provides the methods for a transaction with the key-value store.
+ *
+ * In general: users may assume that if maps (including sorted maps) keyed on byte[] are returned to the user,
+ * they may be accessed, including via random access, via any byte array that is equivalent in terms of
+ * {@link java.util.Arrays#equals(byte[], byte[])}.
+ *
  * @see TransactionManager
  */
 public interface Transaction {
 
+    /**
+     * Returns a mapping of rows to {@link RowResult}s within {@code tableRef} for the specified {@code rows}, loading
+     * columns according to the provided {@link ColumnSelection}.
+     *
+     * The returned {@link SortedMap} is sorted on the byte order of row keys; the ordering of the input parameter
+     * {@code rows} is irrelevant.
+     *
+     * If there are rows with no cells matching the provided {@link ColumnSelection}, they will not be present in the
+     * {@link Map#keySet()} of the output map at all. This accounts for writes and deletes done in this transaction:
+     * a row written to in this transaction will be present, and a row which is deleted in this transaction will be
+     * absent.
+     *
+     * @param tableRef table to load rows from
+     * @param rows rows to be loaded
+     * @param columnSelection columns to load from the given rows
+     * @return a mapping of rows to the columns matching the provided column selection
+     */
     @Idempotent
     SortedMap<byte[], RowResult<byte[]>> getRows(
             TableReference tableRef,
             Iterable<byte[]> rows,
             ColumnSelection columnSelection);
 
+    /**
+     * Returns a mapping of requested {@code rows} to corresponding columns from the queried table.
+     * Only columns matching the provided predicate will be returned, and the single predicate provided applies across
+     * all of the rows.
+     *
+     * The returned {@link BatchingVisitable}s are guaranteed to return cells matching the predicate. These are sorted
+     * by column, with ascending byte ordering.
+     *
+     * It is guaranteed that the {@link Map#keySet()} of the returned map has a corresponding element for each of the
+     * input {@code rows}, even if there are rows where no columns match the predicate.
+     *
+     * @param tableRef table to load values from
+     * @param rows unique rows to apply the column range selection to
+     * @param columnRangeSelection range of columns and batch size to load for each of the rows provided
+     * @return a mapping of rows to cells matching the predicate in the row, following the ordering outlined above
+     * @throws IllegalArgumentException if {@code rows} contains duplicates
```

Comment:
Where is this checked in the code?

Reply:
This is in the builder of the ImmutableSortedMap. I'm not sure if there are existing tests, I added one.